parlai/tasks/empathetic_dialogues/agents.py

```diff
@@ -181,98 +117,40 @@ def _setup_data(self, base_datatype):
 
                 # We've finished the previous episode, so add it to the data
                 turn_idx = 1
-                self.data += self._select_dialogues_to_add(
+                data += self._select_dialogues_to_add(
                     experiencer_text_dialogue, responder_text_dialogue
                 )
                 experiencer_text_dialogue = []
                 responder_text_dialogue = []
 
         # Add in the final episode
-        self.data += self._select_dialogues_to_add(
+        data += self._select_dialogues_to_add(
             experiencer_text_dialogue, responder_text_dialogue
         )
 
+        for episode in data:
```

Comment:
in an ideal world we would make this entire function an iterator that didn't explicitly load data into memory, which speeds up `parlai dd` and lowers peak memory usage.

it's fine for this PR